src/game/overlay.c

```diff
@@ -78,14 +79,105 @@ static RGB888 m_ColorBarMap[][COLOR_STEPS] = {
       { 120, 60, 70 } },
 };
 
-static void Overlay_GetBarLocation(
-    int8_t bar_location, int32_t width, int32_t height, int32_t *x, int32_t *y);
-static void Overlay_DrawBar(int32_t value, int32_t value_max, int32_t bar_type);
+typedef struct BAR_INFO {
+    BAR_TYPE type;
+    int32_t value;
+    int32_t max_value;
+    int32_t percent;
+    bool show;
+    T1M_BAR_SHOW_MODE show_mode;
+    bool blink;
+    int32_t blink_counter;
+    int32_t timer;
+    int32_t color;
+    T1M_BAR_LOCATION bar_location;
+} BAR_INFO;
+
+static BAR_INFO m_HealthBar = { 0 };
+static BAR_INFO m_AirBar = { 0 };
+static BAR_INFO m_EnemyBar = { 0 };
+
+static void Overlay_BarSetupHealth();
+static void Overlay_BarSetupAir();
+static void Overlay_BarSetupEnemy();
+static void Overlay_BarBlink(BAR_INFO *bar_info);
+static void Overlay_BarGetPercent(BAR_INFO *bar_info);
+static void Overlay_BarGetLocation(
+    T1M_BAR_LOCATION bar_location, int32_t width, int32_t height, int32_t *x,
+    int32_t *y);
+static void Overlay_BarDraw(BAR_INFO *bar_info);
 static void Overlay_OnAmmoTextRemoval(const TEXTSTRING *textstring);
 static void Overlay_OnFPSTextRemoval(const TEXTSTRING *textstring);
 
-static void Overlay_GetBarLocation(
-    int8_t bar_location, int32_t width, int32_t height, int32_t *x, int32_t *y)
+static void Overlay_BarSetupHealth()
+{
+    m_HealthBar.type = BT_LARA_HEALTH;
+    m_HealthBar.value = LARA_HITPOINTS;
+    m_HealthBar.max_value = LARA_HITPOINTS;
+    m_HealthBar.percent = 100;
+    m_HealthBar.show_mode = g_Config.healthbar_showing_mode;
+    m_HealthBar.show = false;
+    m_HealthBar.blink = false;
+    m_HealthBar.blink_counter = 0;
+    m_HealthBar.timer = 40;
+    m_HealthBar.color = g_Config.healthbar_color;
+    m_HealthBar.bar_location = g_Config.healthbar_location;
+}
+
+static void Overlay_BarSetupAir()
+{
+    m_AirBar.type = BT_LARA_AIR;
+    m_AirBar.value = LARA_AIR;
+    m_AirBar.max_value = LARA_AIR;
+    m_AirBar.percent = 100;
+    m_AirBar.show_mode = g_Config.airbar_showing_mode;
+    m_AirBar.show = false;
+    m_AirBar.blink = false;
+    m_AirBar.blink_counter = 0;
+    m_AirBar.timer = 0;
+    m_AirBar.color = g_Config.airbar_color;
+    m_AirBar.bar_location = g_Config.airbar_location;
+}
+
+static void Overlay_BarSetupEnemy()
+{
+    m_EnemyBar.type = BT_ENEMY_HEALTH;
+    m_EnemyBar.value = 0;
+    m_EnemyBar.max_value = 0;
+    m_EnemyBar.percent = 100;
+    m_EnemyBar.show_mode = g_Config.healthbar_showing_mode;
+    m_EnemyBar.show = g_Config.enable_enemy_healthbar;
+    m_EnemyBar.blink = false;
+    m_EnemyBar.blink_counter = 0;
+    m_EnemyBar.timer = 0;
+    m_EnemyBar.color = g_Config.enemy_healthbar_color;
+    m_EnemyBar.bar_location = g_Config.enemy_healthbar_location;
+}
+
+static void Overlay_BarGetPercent(BAR_INFO *bar_info)
+{
+    bar_info->percent = bar_info->value * 100 / bar_info->max_value;
```

Comment:
To continue my earlier comment – here I'd prefer for us to just return the calculated value instead of store it in the `BAR_INFO`.